### PR TITLE
added proxy support for build

### PR DESCRIPTION
**Pull request by relief-melone**
As stated in https://github.com/microsoft/vscode-js-debug/issues/1955 running `npx gulp vsDebugServerBundle` behind a proxy will currently fail as got does not honor proxy env vars by default.
This PR addresses that issue by using an https proxy agent in case env vars have been detected for an https proxy

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "execa": "^5.1.1",
     "glob-stream": "^8.0.0",
     "got": "^11.8.6",
+    "https-proxy-agent": "^7.0.4",
```
Comment (connor4312) [inline package.json:70]: This should be in the devDependencies

File: gulpfile.js [hunk 2]
```diff
@@ -351,7 +352,14 @@ gulp.task('package:createVSIX', () =>
 );
 
 gulp.task('l10n:bundle-download', async () => {
-  const res = await got('https://github.com/microsoft/vscode-loc/archive/main.zip').buffer();
+  const opts = {};
+  const proxy = process.env.https_proxy || process.env.HTTPS_PROXY || null;
```
[the remark this answers is not in the document]
Comment (relief-melone) [reply gulpfile.js:356]: this is also running on linux (e.g. in my case I need it for my nvim lsp).

[the remark this answers is not in the document]
Comment (relief-melone) [reply gulpfile.js:356]: > `process.env` is already 'magically' case insensitive on Windows

https://github.com/microsoft/vscode-js-debug/pull/1965

[the remark this answers is not in the document]
Comment (connor4312) [reply gulpfile.js:356]: The uppercase version is the standard one I've seen, and is what e.g. curl uses

**Comment by relief-melone** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

**Comment by relief-melone** [comment]
not a mac user. Any idea what's the catch in that test stage?

File: gulpfile.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (relief-melone) [reply gulpfile.js:356]: don't really have a strong attachment to supporting upper and lowercase and I usually also prefer uppercase env vars. but I have seen multiple programs in linux that only use the lowercase variant. And in linux the lowercase variant also takes precedent over the uppercase one in curl.

**Comment by connor4312** [comment]
That's a flake on main I need to fix still

**Review by connor4312 — APPROVED** [review]

**Review by mjbvz — APPROVED** [review]

**Pull request merged**